ci-operator/step-registry/gcp/provision/minimal-permission/gcp-provision-minimal-permission-commands.sh

```diff
@@ -26,7 +30,48 @@ fi
 # The IAM service account for IPI: ipi-min-permissions-sa@${GOOGLE_PROJECT_ID}.iam.gserviceaccount.com
 # The IAM service account for UPI: upi-min-permissions-sa@${GOOGLE_PROJECT_ID}.iam.gserviceaccount.com
 # Currently we only deal with IPI in Prow CI.
-iam_account="ipi-min-permissions-sa@${GOOGLE_PROJECT_ID}.iam.gserviceaccount.com"
 
-gcloud iam service-accounts keys create "${SHARED_DIR}/gcp_min_permissions.json" --iam-account="${iam_account}" || exit 1
-echo "$(date -u --rfc-3339=seconds) - Created a temporary key of the IAM service account for the minimum permissions testing on GCP."
+if [[ "${MINIMAL_PERMISSIONS_WITHOUT_ACT_AS}" == "no" ]]; then
+  iam_account="ipi-min-permissions-sa@${GOOGLE_PROJECT_ID}.iam.gserviceaccount.com"
+else
+  iam_account="ipi-min-perm-without-actAs-sa@${GOOGLE_PROJECT_ID}.iam.gserviceaccount.com"
+fi
+
+email=$(gcloud --project "${GOOGLE_PROJECT_ID}" iam service-accounts list --filter="email=${iam_account}" --format='value(email)')
+if [[ -z "${email}" ]]; then
+  echo "$(date -u --rfc-3339=seconds) - Failed to find the IAM service account '${iam_account}' in GCP project '${GOOGLE_PROJECT_ID}', abort."
+  exit 1
+fi
+
+# There is a pre-configured key for the IPI minimal permissions testing 
+# service account in QE's project.
+# But no pre-configured key in DEV's project, or for the service account 
+# lack of the permission 'iam.serviceAccounts.actAs' in QE's project.
+if [[ "${MINIMAL_PERMISSIONS_WITHOUT_ACT_AS}" == "no" ]] && [[ -f "${CLUSTER_PROFILE_DIR}/ipi-min-permissions-sa.json" ]]; then
```

Comment:
So we do not have plan to save a separate sa json for `ipi-min-perm-without-actAs-sa` in `CLUSTER_PROFILE_DIR` vault ? For `ipi-min-perm-without-actAs-sa`  case, we always create key for it, right ?

Reply:
I do not think we need to take the way of creating key into consideration for now. We can:
```
sa_filename="ipi-min-permissions-sa.json"
if [[ "${MINIMAL_PERMISSIONS_WITHOUT_ACT_AS}" == "yes" ]]; then
    sa_filename="ipi-min-perm-without-actAs-sa.json"
fi
cp "${CLUSTER_PROFILE_DIR}/${sa_filename}" "${SHARED_DIR}/gcp_min_permissions.json"
```

Reply:
Updated.

Reply:
Now I think we can keep consistent way - saving sa in vault, rather than if create key or not in different cases.